docs/tutorials/gpu_executor.md

```diff
@@ -0,0 +1,302 @@
+
+If you need to have CUDA installed in the image, you usually have two options: either you take the `nvidia/cuda` for the base image, or you take the official GPU-enabled image of the framework you are using, for example, `tensorflow/tensorflow:2.6.0-gpu`.
+```
+
+The other file we care about in this case is `config.yml`, and here the default version works as well. So let's build the docker image
+
+```bash
+docker build -t sentence-encoder .
+```
+
+You can run the container to quickly check that everything is working well
+
+```bash
+docker run sentence-encoder
+```
+
+Now, let's use the docker version of our encoder with the GPU. If you've dealt with GPUs in containers before, you probably remember that to use a GPU insite the container you need to pass `--gpus all` command to `docker run`. And jina enables you to do just that.
+
+Here's how we need to modify our `main.py` script to use a GPU-base docker executor
+
+```python
+from jina import Document, DocumentArray, Flow
+
+from executor import SentenceEncoder
+
+def generate_docs():
+    for _ in range(10000):
+        yield Document(
+            text='Using a GPU enables you to significantly speed up encoding'
+        )
+
+f = Flow().add(
+    uses='docker://sentence-encoder', uses_with={'device': 'cuda'}, gpus='all'
+)
+with f:
+    f.post(on='/encode', inputs=generate_docs, show_progress=True, request_size=32)
+```
+
+If we run this with `python main.py`, we'll get the same output as before, except that now we'll also get the output from the Docker container.
+
+You may notice that every time we start the executor, the transformer model gets downloaded again. To speed this up, we would want the encoder to load the model from a file which we have pre-downloaded to our disk.
+
+We can do this with docker volumes - jina will simply pass the argument to the Docker container. Here's how we modify the `main.py` to allow that
+
+```python
+f = Flow().add(
+    uses='docker://sentence-encoder',
+    uses_with={'device': 'cuda'},
+    gpus='all',
+    # This has to be an absolute path, replace /home/ubuntu with your home directory
+    volumes="/home/ubuntu/.cache:/root/.cache",
+)
+```
+
+Here we mounted the `~/.cache` directory, because this is where pre-built transformer models are saved in our case. But this could also be any custom directory - depends on the python package you are using, and how you specify the model loading path.
+
+Now, if we run `python main.py` again you can see that no downloading happens inside the container, and that the encoding starts faster.
+
+## Using GPU with Hub executors
+
+We now saw how to use GPU with our executor locally, and when using it in a Docker container. What about when we use executors from the Hub, is there any difference?
+
+Nope! Not only that, many of the executors on the hub already come with a GPU-enabled version pre-built, usually under the `gpu` tag. Let's modify our example to use the pre-built `TransformerTorchEncoder` from the Hub
```

Comment:
If we have a section about tags in jina hub documentation, let's mention it. Otherwise, maybe we can create a sectionto describe how to use tags ?

Reply:
I'll check if there is one - if not, I would do this in another PR